tests/test_crypto.py

```diff
@@ -69,3 +81,98 @@ def test_subprocess_raises_exception(safe_tmpdir, mocker):
         gpg.decrypt_submission_or_reply(test_gzip, output_filename, is_doc=True)
 
     assert mock_gpg.call_count == 1
+
+
+def test_import_key(safe_tmpdir, config, source):
+    '''
+    Using the `config` fixture to ensture the config is written to disk.
+    '''
+    helper = GpgHelper(str(safe_tmpdir), is_qubes=False)
+    helper.import_key(source['uuid'], source['public_key'])
```

Comment:
there's a platform independence issue: gpg has [a limit to the length of its homedir](https://lists.gnupg.org/pipermail/gnupg-users/2017-February/057692.html), and with the individual temporary folder we have for tests, we hit that limit on (at least) macOS:

```
-------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------
crypto.py                  141 ERROR    Could not import key: Command '['gpg', '--homedir', '/private/var/folders/gv/2r10zvfj303fyd5_6yn_p0b80000gn/T/pytest-of-redshiftzero/pytest-8/test_encrypt_fail0/gpg', '--trust-model', 'always', '--import-options', 'import-show', '--with-colons', '--import', '/var/folders/gv/2r10zvfj303fyd5_6yn_p0b80000gn/T/tmpsddsl_5h']' returned non-zero exit status 2
gpg: keybox '/private/var/folders/gv/2r10zvfj303fyd5_6yn_p0b80000gn/T/pytest-of-redshiftzero/pytest-8/test_encrypt_fail0/gpg/pubring.kbx' created
gpg: key 6179D97BCFA52E5F: public key "securedropclient-test <test@securedrop.org>" imported
gpg: can't connect to the agent: File name too long
gpg: Total number processed: 1
gpg:               imported: 1
```

I actually hit this before the alpha deadline when I was trying to use `python-gnupg` to do the file decryption, and the workaround I came up with was to use `tempfile.TemporaryDirectory()` : https://github.com/freedomofpress/securedrop-client/compare/client-decrypt-files#diff-e35a4bac9b30adda70e9a2f001ea05d4R12 which has a sufficiently short path.... 🤷‍♀️ 

I recognize that this means that this messes up e.g. your `config` fixture and means we would also need to change them. I'm happy to do so in a minimal commit on the end here if you like since it's macOS only, let me know.

Reply:
Wow that is a whack bug to have hit. I can make that change when I polish up the reset of the things. I'll probably just call it `homedir` and create all the dirs we need to avoid cluttering the tests with a bunch of `os.mkdir`s.